src/components/RangeInput/RangeInput.tsx

```diff
@@ -1,49 +1,62 @@
 /** @jsx h */
 
 import { h, Component } from 'preact';
-import PropTypes from 'prop-types';
 import cx from 'classnames';
 import Template from '../Template/Template';
+import {
+  RangeInputCSSClasses,
+  RangeInputTemplates,
+} from '../../widgets/range-input/range-input';
+import { Range, RangeBoundaries } from '../../connectors/range/connectRange';
+
+export type RangeInputComponentCSSClasses = {
+  [TClassName in keyof RangeInputCSSClasses]: string;
+};
 
-class RangeInput extends Component {
-  constructor(props) {
-    super(props);
+export type RangeInputProps = {
+  min?: number;
+  max?: number;
+  step: number;
+  values: Partial<Range>;
+  cssClasses: RangeInputComponentCSSClasses;
+  templateProps: {
+    templates: RangeInputTemplates;
+  };
+  refine(rangeValue: RangeBoundaries): void;
+};
 
-    this.state = {
-      min: props.values.min,
-      max: props.values.max,
-    };
-  }
+class RangeInput extends Component<RangeInputProps, Partial<Range>> {
+  public state = {
+    min: this.props.values.min,
+    max: this.props.values.max,
+  };
 
-  componentWillReceiveProps(nextProps) {
+  public componentWillReceiveProps(nextProps: RangeInputProps) {
     this.setState({
       min: nextProps.values.min,
       max: nextProps.values.max,
     });
   }
 
-  onInput = name => event => {
+  private onInput = (key: string) => (event: Event) => {
```

Comment:
```suggestion
  private onInput = (key: string) => (event: InputChangeEvent) => {
```

not sure what the right name is, but I think there's a more precise event that can avoid the cast

Reply:
I think the event you think of is the `ChangeEvent` but it's in React, I don't see any other event to do it, let me know if you found the name!

Reply:
Since we use this implementation in our codebase (e.g. [SearchBox](https://github.com/algolia/instantsearch.js/blob/master/src/components/SearchBox/SearchBox.tsx#L78) and it's as correct as the `preact` implementation, I think we should stick to it!